src/Monolog/Formatter/MongoDBFormatter.php

```diff
@@ -105,6 +105,12 @@ protected function formatException(\Throwable $exception, int $nestingLevel)
 
     protected function formatDate(\DateTimeInterface $value, int $nestingLevel): UTCDateTime
     {
-        return new UTCDateTime((string) floor($value->format('U.u') * 1000));
+        $miliseconds = floor($value->format('U.u') * 1000);
+
+        $miliseconds = version_compare(phpversion('mongodb'), '1.1.9', '>')
```

Comment:
For versions after 1.1.9 this is correct, but needs refining for versions before 1.1.9: According to [the constructor](https://github.com/mongodb/mongo-php-driver/blob/1.1.9/src/BSON/UTCDateTime.c#L61..L83), on 64-bit systems it would parse the parameter as an integer while it expects a string on 32-bit systems. So, when running this code on a 32-bit system with `strict_types` enabled it would still error out. You can use the `PHP_INT_SIZE` constant to figure out the correct behavior for older driver versions.